src/main/java/sirius/biz/jobs/batch/file/DictionaryBasedImportJob.java

```diff
@@ -61,6 +63,7 @@ protected FieldDefinition customFieldLookup(String field) {
     @Override
     public void handleRow(int index, Values row) {
         if (index == 1) {
+            dictionary = determineDictionary();
```

Comment:
sollte das wirklich das lokal gespeicherte dictionary überschreiben? ich finde das potentiell gefährlich

Reply:
Es wird nur ein dictionary benutzt gleichzeitig. Meistens jemand die das überschreibt gibt kein dictionary by init weil welchen per File definiert sind. Hat den vorteil, dass wenn man vergisst ein dict mitzugeben, zb für ein neuen Datei, dass es nicht der letzte verwendet.  Aber ja, Design frage...
#open4suggestions :)

Reply:
Meine Suggestion wäre halt in der Methode in der aufs dictionary zugegriffen wird determineDictionary aufzurufen und wenn das nichts zurück gibt auf das this.dictionary zurückzugreifen. Oder wenns mehrere stellen sind eine private/protected methode einzuführen in der die logik liegt.

Felder die je nach State überschrieben werden finde ich nicht ganz so schön bzw halt einfach nicht so klar überblickbar was in welchem Moment in dem Feld stehen könnte.

Reply:
Hab nicht so gemacht weil handleRow wird pro Zeile aufgerufen, der Dictionary wird in die 1. Zeile initialisiert. Damit braucht man noch einen Klassenvar wie currentDictionary.

Reply:
Nope, einfach nicht an die globale variable binden. Warte ich mach ne Suggestion :)

Reply:
Ah doch wegen dem if, also nur in Zeile 1 ... Ich finds trotzdem nicht ganz so schön